### PR TITLE
feat: tasks view tree restructure - support to multi root workspaces

### PR DESCRIPTION
**Pull request by alex-gilin**

### CODE REVIEW

File: packages/tasks_panel/src/utils/ws-folder.ts [hunk 1]
```diff
@@ -7,3 +7,15 @@ export function cleanTasks(tasks: ConfiguredTask[]): void {
     delete _.index;
   });
 }
+
+export const DEPLOY = "Deploy";
+export const BUILD = "Build";
+export const MISC = "Miscellaneous";
+
+export function isMatchDeploy(intent: string): boolean {
+  return /^deploy.*$/gi.test(intent);
```
Comment (veredcon) [inline packages/tasks_panel/src/utils/ws-folder.ts:16]: is it OK to check if it exactly starts with "build"/"deploy"? or is it enough that it includes it somewhere?

File: packages/tasks_panel/src/view/task-tree-item.ts [hunk 2]
```diff
@@ -1,27 +1,47 @@
 import { find } from "lodash";
-import { TreeItem, TreeItemCollapsibleState, Command, tasks } from "vscode";
+import { TreeItem, TreeItemCollapsibleState, Command, tasks, ThemeIcon } from "vscode";
+import { isMatchBuild, isMatchDeploy } from "../../src/utils/ws-folder";
 
 type TaskStatus = "idle" | "running";
 
-export class IntentTreeItem extends TreeItem {
-  constructor(public label: string, public collapsibleState: TreeItemCollapsibleState) {
+class BranchTreeItem extends TreeItem {
```
Comment (veredcon) [inline packages/tasks_panel/src/view/task-tree-item.ts:7]: Why is it called "BranchTreeItem"? what it represents?

File: packages/tasks_panel/src/view/tasks-tree.ts [hunk 3]
```diff
@@ -23,26 +33,59 @@ export class TasksTree implements TreeDataProvider<TreeItem> {
...
+    const intents = sortBy(uniq(map(this.filterByFolder(tasks, parent), "__intent")));
+    getClassLogger(LOGGER_CLASS_NAME).debug(messages.GET_TREE_BRANCHES("intent", intents.length));
+    return map(intents, (_) => new IntentTreeItem(_, TreeItemCollapsibleState.Collapsed, parent));
+  }
+
+  private getWorkspaces(wsFolders: string[]): IntentTreeItem[] {
+    getClassLogger(LOGGER_CLASS_NAME).debug(messages.GET_TREE_BRANCHES("workspace", wsFolders.length));
+    return map(
+      wsFolders,
+      (_) =>
```
Comment (veredcon) [inline packages/tasks_panel/src/view/tasks-tree.ts:50]: why you are using "_" and not a variable name (e.g. wsFolderPath)?

File: packages/tasks_panel/src/view/tasks-tree.ts [hunk 4]
```diff
@@ -23,26 +33,59 @@ export class TasksTree implements TreeDataProvider<TreeItem> {
...
+          workspace.getWorkspaceFolder(Uri.file(_))?.name ?? "",
+          _,
+          TreeItemCollapsibleState.Collapsed
+        )
+    );
+  }
+
+  private getRoots(tasks: ConfiguredTask[]): TreeItem[] {
+    const wsFolders = uniq(map(tasks, "__wsFolder"));
+    return wsFolders.length === 1 ? this.getIntents(tasks) : this.getWorkspaces(wsFolders);
```
Comment (veredcon) [inline packages/tasks_panel/src/view/tasks-tree.ts:61]: why if there is only why ws folder it does not appear as root? is it by definition?

**Review by veredcon — APPROVED** [review]

File: packages/tasks_panel/src/utils/ws-folder.ts (hunk 1 above)
Comment (alex-gilin) [reply packages/tasks_panel/src/utils/ws-folder.ts:16]: this is the exact definition

File: packages/tasks_panel/src/view/task-tree-item.ts (hunk 2 above)
Comment (alex-gilin) [reply packages/tasks_panel/src/view/task-tree-item.ts:7]: it represent a common branch in tasks tree. The `ProjectTreeItem` and `IntentTreeItem` extends it with its own properties.

File: packages/tasks_panel/src/view/tasks-tree.ts (hunk 3 above)
Comment (alex-gilin) [reply packages/tasks_panel/src/view/tasks-tree.ts:50]: changed

File: packages/tasks_panel/src/view/tasks-tree.ts (hunk 4 above)
Comment (alex-gilin) [reply packages/tasks_panel/src/view/tasks-tree.ts:61]: yes. this is exact definition.

**Review by veredcon — APPROVED** [review]

**Pull request merged**